GVFS/GVFS.FunctionalTests/Tools/GVFSProcess.cs

```diff
@@ -51,12 +55,28 @@ public bool TryMount(out string output)
 
         public string AddSparseFolders(params string[] folders)
         {
-            return this.CallGVFS($"sparse {this.enlistmentRoot} -a {string.Join(";", folders)}");
+            return this.SparseCommand(addFolders: true, shouldSucceed: true, folders: folders);
+        }
+
+        public string AddSparseFolders(bool shouldSucceed, params string[] folders)
+        {
+            return this.SparseCommand(addFolders: true, shouldSucceed: shouldSucceed, folders: folders);
         }
 
         public string RemoveSparseFolders(params string[] folders)
         {
-            return this.CallGVFS($"sparse {this.enlistmentRoot} -r {string.Join(";", folders)}");
+            return this.SparseCommand(addFolders: false, shouldSucceed: true, folders: folders);
+        }
+
+        public string RemoveSparseFolders(bool shouldSucceed, params string[] folders)
+        {
+            return this.SparseCommand(addFolders: false, shouldSucceed: shouldSucceed, folders: folders);
+        }
+
+        public string SparseCommand(bool addFolders, bool shouldSucceed, params string[] folders)

```

Comment:
Are there any code paths where this will be called with `shouldSucceed` set to `false`?  Or will this feature be used in a subsequent commit?

Reply:
Yes see line 206, 255 in SparseTests.